docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,726 @@
+contexts:
+- context:
+    cluster: local
+    user: kubelet
+  name: service-account-context
+current-context: service-account-context
+```
+Put the kubeconfig(s) on every node.  The examples later in this
+guide assume that there are kubeconfigs in `/var/lib/kube-proxy/kubeconfig` and
+`/var/lib/kubelet/kubeconfig`.
+
+## Configuring and Installing Base Software on Nodes
+
+This section discusses how to configure machines to be kubernetes nodes. 
+
+You should run three daemons on every node:
+  - docker or rkt
+  - kubelet
+  - kube-proxy
+
+You will also need to do assorted other configuration on top of a
+base OS install.
+
+Tip: One possible starting point is to setup a cluster using an existing Getting
+Started Guide.   After getting a cluster running, you can then copy the init.d scripts or systemd unit files from that
+cluster, and then modify them for use on your custom cluster.
+
+### Docker
+The minimum required Docker version will vary as the kubelet version changes.  The newest stable release is a good choice.  Kubelet will log a warning and refuse to start pods if the version is too old, so pick a version and try it.
+
+If you previously had Docker installed on a node without setting Kubernetes-specific
+options, you may have a Docker-created bridge and iptables rules.  You may want to remove these
+as follows before proceeding to configure Docker for Kubernetes.
+```
+iptables -t nat -F 
+ifconfig docker0 down
+brctl delbr docker0
+```
+
+The way you configure docker will depend in whether you have chosen the routable-vip or overlay-network approaches for your network.
+Some docker options will want to think about:
+  - `--bridge cbr0`
+  - `--iptables=false`
+    - so that kube-proxy can manage iptables instead of docker.
+  - `--ip-masq=false`
+    - if you have setup PodIPs to be routable, then you want this false, otherwise, docker will
+      rewrite the PodIP source-address to a NodeIP.
+      cluster.
+    - if you are using an overlay network, consult those instructions.  
+  - `--bip=`
+    - should be the CIDR range for pods for that specific node.
+  - `--mtu=`
+    - may be required when using Flannel, because of the extra packet size due to udp encapsulation
+  - `--insecure-registry $CLUSTER_SUBNET` 
+    - to connect to a private registry, if you set one up, without using SSL.
+
+You may want to increase the number of open files for docker:
+   - `DOCKER_NOFILE=1000000`
+
+Ensure docker is working correctly on your system before proceeding with the rest of the
+installation, by following examples given in the Docker documentation.
+
+### rkt
+
+[rkt](https://github.com/coreos/rkt) is an alterative to Docker.  You only need to install one of Docker or rkt.
+
+*TODO*: how to install and configure rkt.
+
+### kubelet
+
+All nodes should run kubelet.  See [Selecting Binaries](#selecting-binaries).
+
+Arguments to consider:
+  - If following the HTTPS security approach:
+    - `--api_servers=https://$MASTER_IP`
```

Comment:
spec flags with dashes rather than underscores?  Our daemons should all support both, but dashes are the "right" way